game_states.py

```diff
@@ -0,0 +1,189 @@
+import pygame
+import pymlgame
+import time
+
+from misc import Point
+
+
+class PygameSurfaceDecorator:
+    """
+    A wrapper class around pygame surfaces that monkey patches differences
+    between pymlgame and pygame
+    """
+
+    def __getattribute__(self, item):
+        """
+        method that tries to first return method of decorator.
+        If that method is not present this methods tries to return
+        the property from the decorated object.
+        """
+        try:
+            v = object.__getattribute__(self, item)
+        except AttributeError:
+            v = getattr(object.__getattribute__(self, 'surface'), item)
+        return v
+
+    def __init__(self, surface):
+        self.surface = surface
+
+    @property
+    def width(self):
+        return self.surface.get_width()
+
+    @property
+    def height(self):
+        return self.surface.get_height()
+
+    @property
+    def matrix(self):
+        return pygame.surfarray.array3d(self.surface)
+
+
+class GameState:
+    """
+    Abstract base class for a state of a pyngpong game. Derived classes should
+    implement update() and render().
+    States will go ahead and replace themselves by another state if they deem
+    it necessary.
+    """
+
+    def __init__(self, game):
+        self.game = game
+
+    def _init_font_rendering(self):
+        if not pygame.font.get_init():
+            pygame.font.init()
+
+        self.font = pygame.font.Font('assets/fonts/6px2bus.ttf', 6)
+
+    def update(self):
+        pass
+
+    def render(self):
+        pass
+
+
+class WaitingState(GameState):
+    """
+    In this state, the game waits for 2 players to join.
+    """
+
+    def __init__(self, *args,  **kwargs):
+        super(WaitingState, self).__init__(*args, **kwargs)
```

Comment:
Care to modernize the other subclasses `__init__` methods as well? Didn't know the python3 way.